### PR TITLE
chore(manifest): parse container dependencies 

### PR DESCRIPTION
**Pull request by KollaAdithya**
This is first PR which will parse the container dependencies from manifest.
In the next PR - based on these order of dependencies start and stop the containers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/run_local.go [hunk 1]
```diff
@@ -515,9 +515,31 @@ func (o *runLocalOpts) prepareTask(ctx context.Context) (orchestrator.Task, erro
...
+func (o *runLocalOpts) containerDependencies(unmarshaledManifest interface{}) map[string]manifest.ContainerDependency {
+	type containerDependency interface {
+		ContainerDependencies() map[string]manifest.ContainerDependency
+	}
+	mf, ok := unmarshaledManifest.(containerDependency)
+	if ok {
+		return mf.ContainerDependencies()
+	}
+	return nil
+}
```
Comment (CaptainCarpensir) [inline internal/pkg/cli/run_local.go]: Does this serve any purpose to be a method of `runLocalOpts`?

File: internal/pkg/manifest/lb_web_svc_test.go [hunk 2]
```diff
@@ -2838,6 +2838,71 @@ func TestLoadBalancedWebService_BuildArgs(t *testing.T) {
 	}
 }
 
+func TestLoadBalancedWebService_ContainerDependencies(t *testing.T) {
+	testCases := map[string]struct {
+		in                 *LoadBalancedWebService
+		wantedDependencies map[string]ContainerDependency
+		wantedErr          error
+	}{
+		"return container dependencies of all containers": {
```
Comment (CaptainCarpensir) [inline internal/pkg/manifest/lb_web_svc_test.go:2847]: Can you include a test case for the firelens logging container?

**Review by CaptainCarpensir — COMMENTED** [review]
Looks great!

File: internal/pkg/cli/run_local.go (hunk 1 above)
Comment (KollaAdithya) [reply internal/pkg/cli/run_local.go]: Good catch we can move this to `cli/deploy/workload.go`

**Review by CaptainCarpensir — APPROVED** [review]
LGTM!

File: internal/pkg/cli/deploy/workload.go [hunk 3]
```diff
@@ -738,6 +738,18 @@ func envFiles(unmarshaledManifest interface{}) map[string]string {
 	return nil
 }
 
+// ContainerDependencies returns a map of ContainerDependency objects from workload manifest.
+func ContainerDependencies(unmarshaledManifest interface{}) map[string]manifest.ContainerDependency {
```
Comment (iamhopaul123) [inline internal/pkg/cli/deploy/workload.go]: We might want to move this to the `manifest` pkg. Any reason we want to keep it here?

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**